accounts/abi/unpack_test.go

```diff
@@ -357,21 +357,21 @@ var unpackTests = []unpackTest{
 	},
 	{
 		def: `[{"name":"int_one","type":"int256"}]`,
-		enc: "00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000002",
+		enc: "0000000000000000000000000000000000000000000000000000000000000001",
 		want: struct {
 			IntOne *big.Int
 		}{big.NewInt(1)},
 	},
 	{
 		def: `[{"name":"int__one","type":"int256"}]`,
-		enc: "00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000002",
+		enc: "0000000000000000000000000000000000000000000000000000000000000001",
 		want: struct {
 			IntOne *big.Int
 		}{big.NewInt(1)},
 	},
 	{
 		def: `[{"name":"int_one_","type":"int256"}]`,
-		enc: "00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000002",
+		enc: "0000000000000000000000000000000000000000000000000000000000000001",
```

Comment:
I don't see why you remove the trailing uint256? From what I gather, this is intended to make sure that if there is more than one int, `unpack` will only decode the first one. Now it is possible that the boundary is checked and that you return an error, but in that case a corresponding test needs to be added.